### ISSUE TITLE
Change in column "types" as a result of applying column filters 

### ISSUE THREAD

**Issue by hzarkoob** [rstudio/rstudio#5479]
### System details

    RStudio Edition : Server
    RStudio Version : 1.2.1578 (latest RStudio preview)
    OS Version      : Server, Ubuntu 16.04.6 LTS, Xenial
    R Version       : 3.6.0

### Steps to reproduce the issue:
- Import a csv file by clicking on it in the file pane.
- Change the default _type_ of one of the columns while the dataset is being imported.
- Once the dataset is imported and (automatically) viewed, use the column filter to select a subset of columns using their indices – including the column whose _type_ was customized.
- It can be seen that the column _type_ no longer matches the selected (customized) type. (It seems like it goes back to the automatically inferred _type_.) The way I could tell that was the following. I forced a column whose values looked _integer_ to be imported as a _character_ column. Later, when I tried to apply a row-filter to that column, a histogram was shown which signals the column _type_ was assumed to be numeric.

**Comment by ronblum** [comment]
@hzarkoob Thank you for raising this issue. I'm unable to reproduce it, though. Can you check the field types in the object, using `str()`?

**Comment by hzarkoob** [comment]
@ronblum After spending some more time on this, I noticed the following more straightforward example better reflects on the issue. Please go through the following steps to reproduce the issue:

- Run the following code:

```
testData = data.frame(logicalVar = sample(c(T,F), size = 10, replace = T), doubleVars = matrix(rnorm(500), 10))
View(testData)
```
It will create a data frame where the type of the first column is logical, and the type of the remaining columns is double.

- Adjust indices of columns to see only Columns 50-51 (A total of two columns). Then apply "Filter". You will notice that the type of Column 50 (the first column in the new two-column display) will be Logical, while it should be double.

**Comment by ronblum** [comment]
@hzarkoob Thank you! I can reproduce this in RStudio Server 1.2.5001 on Ubuntu 18.04 and RStudio Desktop 1.2.5001 on MacOS 10.15.

**Comment by dfalty** [comment]
Verified fixed 1.3.822-1
